### PR TITLE
[Rust] Add rust runtime to CI

### PR DESCRIPTION
**Pull request by nhynes**

### CODE REVIEW

**Comment by tqchen** [comment]
If the testcase relies on the change of the ci_cpu docker file, it won't work as the ci images are cached, please send changes to CI image as a separate PR

**Comment by nhynes** [comment]
depends on #1855

**Comment by tqchen** [comment]
I checked the ci image locally and the test does not run. You can try to reproduce locally via
- docker/build.sh ci_cpu bash
- docker/bash.sh ci_cpu 
   - Type cargo and run the test scripts.

The main reason is that default rustup installs the env only for the current user(which is root). And docker/bash.sh script will actually enter an env that is the same as the user who typed the command. And in that case /root/.cargo is not accessible.

Seems some customized configuration is necessary to make sure rust installation works for all users in the system. @nhynes can you try to do some tests locally and send an updated version of script that works? Also the mod of test/script/task_rust.sh need to be changed to executable

**Comment by nhynes** [comment]
seems to work now. thanks for the debugging tips

File: tests/scripts/task_rust.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+export LD_LIBRARY_PATH=lib:${LD_LIBRARY_PATH}
+export PYTHONPATH=python:nnvm/python:topi/python
```
Comment (tqchen) [inline tests/scripts/task_rust.sh]: The test was not successful

NOTE you are using a relative path here. after you cd into rust, the PYTHONPATH is no longer correctly, try to run this script inside the docker

Comment (tqchen) [reply tests/scripts/task_rust.sh]: you need to chmod +x tests/scripts/task_rust.sh

**Comment by tqchen** [comment]
Some further comment. Note that you can verify locally by running the script
```bash
docker/bash.sh tvmai/ci-cpu
./tests/scripts/task_rust.sh
```

**Comment by nhynes** [comment]
Sorry about that. It's 100% for sure actually fixed now.

Also, there's an independent PR that will make rust builds way faster (#1859). That one can be merged after this one since sccache transparently makes things better.

**Comment by tqchen** [comment]
Locally, here is the error message:

```bash
docker/bash.sh tvmai/ci-cpu
# inside the docker
./tests/task_rust.sh

error: failed to load source for a dependency on `bounded-spsc-queue`

Caused by:
  Unable to update registry `https://github.com/rust-lang/crates.io-index`

Caused by:
  failed to open: /opt/rust/registry/index/github.com-1ecc6299db9ec823/.cargo-index-lock

Caused by:
  Permission denied (os error 13)
tqchen@mode:~$ ls /opt/rust/registry/
cache/ index/ src/   
tqchen@mode:~$ ls /opt/rust/registry/index/github.com-1ecc6299db9ec823/.cargo-index-lock 
/opt/rust/registry/index/github.com-1ecc6299db9ec823/.cargo-index-lock

```

**Comment by nhynes** [comment]
That's odd because the installation chmods everything to writable https://github.com/dmlc/tvm/pull/1859/files#diff-8d7a51347f558899828488f5218fb220L15

I'll give it another go.

**Comment by nhynes** [comment]
fixed and tested

**Comment by tqchen** [comment]
Please remove the changes in Jenkinsfile for now, then i will merge this in, rebuild the image and trigger Jenkins through a separate  PR

**Pull request merged**